edx_rest_framework_extensions/tests/test_permission.py

```diff
@@ -27,3 +36,103 @@ def test_has_permission_with_invalid_users(self, user):
         request.user = user
         permission = IsSuperuser()
         self.assertFalse(permission.has_permission(request, None))
+
+
+class JwtHasScopeTests(TestCase):
+    """ Tests for JwtHasScope permission class. """
+    def setUp(self):
+        super(JwtHasScopeTests, self).setUp()
+        self.user = UserFactory()
+        self.scopes = ('grades:read', 'certificates:read')
+
+    @patch('edx_rest_framework_extensions.decorators.waffle.switch_is_active')
+    def test_scopes_validated(self, waffle_mock):
+        """
+        Test that the permission check returns true when scopes are validated.
+        """
+        waffle_mock.return_value = True
+        request = RequestFactory().get('/')
+        request.successful_authenticator = JwtAuthentication()
+        request.auth = generate_jwt(self.user, scopes=self.scopes)
+        view = Mock(required_scopes=self.scopes)
+        self.assertTrue(JwtHasScope().has_permission(request, view))
+
+    def test_scope_enforcement_disabled(self):
+        """
+        Test that the permission check is skipped if scope enforcement is disabled.
+        """
+        request = RequestFactory().get('/')
+        self.assertTrue(JwtHasScope().has_permission(request, None))
+
+    @patch('edx_rest_framework_extensions.decorators.waffle.switch_is_active')
+    def test_not_jwt_authentication(self, waffle_mock):
```

Comment:
Nit: This test may read better and ensure coverage if there was a `ddt` of various values for `successful_authenticator` with their expected returns from `has_permission`.